UI.py

```diff
@@ -347,6 +365,41 @@ def FormatGetEndOfMonthReportEmbed(member, month, year):
         
     return member_data
 
+def CreateGetEndOfYearReportEmbed(members_list, year = None):
+    year = year or datetime.datetime.now().year
+    embed = discord.Embed(
+        title = f'End of Year Report',
+        description = f'{year} Report:',
+        colour = 0x4682B4
+    )
+    embed.set_thumbnail(url = os.getenv("Salary_Image"))
+    embed.add_field(name = '\u200B', value = '\u200B', inline = False)
+
+    for member in members_list:
+        member_data = FormatGetEndOfYearReportEmbed(member, year)
+        embed.add_field(name = f'**{member["name"].upper()}**', value = member_data, inline = False)
+        embed.add_field(name = '\u200B', value = '\u200B', inline = False)
+
+    footer_text = (("\u200B " * embed_footer_spaces_count) + datetime.date.today().strftime("%d/%m/%Y"))
+    embed.set_footer(text = footer_text)
+    return embed
+
+def FormatGetEndOfYearReportEmbed(member, year):
+    member_data = ""
+    start_date = datetime.datetime(year, 1, 1)
+    end_date = datetime.datetime(year, 12, utils.GetMonthDaysCount(12, year))
```

Comment:
we don't start from 1/1, it's the same situation as in the months. you can't drop the last few days of the year before

Reply:
Done